kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -102,59 +103,47 @@ define([
          * Returns a Promise that resolves once the rendering is done.
          */
         render: function () {
-            return this.updateView();
+            // a full render should wipe the entire view.
+            // really, this should only be done once. Subsequent data fetching
+            // just updates the data in the datatable.
+            this.$elem.empty();
+            this.renderFileHeader();
+            return this.updateView().then(() => {
+                this.renderImportButton();
+            });
         },
 
         updateView: function () {
-            return Promise.resolve(
-                this.stagingServiceClient.list({
-                    path: this.subpath,
-                })
-            )
+            return Promise.resolve(this.stagingServiceClient.list())
                 .then((data) => {
-                    //list is recursive, so it'd show all files in all subdirectories. This filters 'em out.
-                    const files = JSON.parse(data).filter((f) => {
-                        // this is less complicated than you think. The path is the username,
-                        // subpath, and name concatenated. The subpath may be empty so we
-                        // filter it out and only join defined things. If that's the same as
-                        // the file's path, we're at the right level. If not, we're not.
-                        return (
-                            [this.userInfo.user, this.subpath, f.name]
-                                .filter((p) => p.length > 0)
-                                .join('/') === f.path
-                        );
-                    });
-                    files.forEach((f) => {
-                        if (!f.isFolder) {
-                            f.imported = {};
-                        }
+                    const files = JSON.parse(data).map((f) => {
```

Comment:
No more filtering, just get the files.
This also adds the subdirectory `subdir` to each file object.
All of the files have a `path` string which looks like:
`username/some_directory/filename.txt`

All of the staging area (and app) functions don't need the username, and that actually causes problems. So just strip it out here. This makes the path
`some_directory/filename.txt`

The `subdir`, then, becomes
`/some_directory`.

Reply:
much simpler!